collections/_blogs/2020-09-19-performance-benchmarking-using-meshery-and-nighthawk.md

```diff
@@ -0,0 +1,75 @@
+---
+layout: post
+title:  "Performance benchmarking using Meshery and Nighthawk"
+date:   2020-09-19 11:00:05 +0530
+image: /assets/images/posts/2020-04-26-getting-started-with-mesheryctl/mesheryctl.png
+author: Kush Trivedi
+permalink: /blog/performance-benchmarking-using-meshery-and-nighthawk
+---
+<div class="center" style="color:gray;position:relative;top:-10px;font-size:1.25em;"><i>Google Summer of Code 2020 with CNCF</i></div>
+
+<h4>What is performance benchmarking and Why should I care?</h4>
+
+Many of you may remember HBO streaming service crashed when the final season of Game of Thrones went live on the service. The streaming service and the website was not able to engage all the user requests and was not robust which ultimately resulted in DoS (Denial of Service). There are many examples of service crashes from time to time which makes us look into performance benchmarking and makes one to care about the product's performance before it is launched in the market.
+
+There are several advantages of early benchmarking of the services which may enhance the reponse time of services, may resolve glitches, can enhance the application's robustness, and bring stability and dependability within an application.
+
+There are several tools for performance benchmarking out of which one of the most important is load generators and benchmarks. Some of the well-known load generators in cloud-native realm :
+- Fortio
+- Wrk2/Wrk
+- Vegeta
+and then there is **Nighthawk**
+
+To sum it all up, there is **Meshery**, service mesh management plane with ability to generate load and benchmark service in/out of the service mesh using Fortio, Wrk2 and now, Nighthawk too.
+
+
+<h4> Meshery + Nighthawk = Robust Distributed and Scalable Services </h4>
+
+
+Nighthawk, written and maintained by Envoy Community is a L7 (HTTP/HTTPS/HTTP2) performance characterization tool which supports performance benchmarking using HTTP  or gRPC service.
+
+Nighthawk is built using open-source build and test tool Bazel and supports output formatting to well-known formats, allowing integration with other systems and dashboards. To add more to the list, it is going to soon support distribute performance benchmarking which is under active development.
+
+On the other hand, Meshery, written and maintainer by Layer5 Community is the multi-service mesh management plane offering lifecycle, configuration and performance management of service meshes and their workloads. With Meshery,
+not only you can manage your service-meshes lifecycle but can also benchmark them with industry grade tools to let you know how much under and over utilised your services are.
+
+Soon, Meshery is going to be a canonical implementation for [Service Mesh Performance Specification](https://github.com/layer5io/service-mesh-performance) which is a common and standardised format to describe :
```

Comment:
```suggestion
Soon, Meshery is going to be a canonical implementation for [Service Mesh Performance](https://github.com/layer5io/service-mesh-performance), which is a common and standardized format to describe :
```

Reply:
The specification is needed here as when you reference SMI you reference it as SMI Spec.